text/0000-wire-reform.md

```diff
@@ -0,0 +1,146 @@
+
+# Summary
+
+This RFC describes the way to decouple the wire service from LWC entirely, and implement reactive tracking for wired configuration and wired methods.
+
+# Motivations
+
+There is a dual-dependency between `@lwc/engine` and `wire-service`, even though neither of those two packages are importing each other, it is the responsibility of the adapter author to connect them via `registerWireService(register)` where `registerWireService()` is provided by `wire-service` and `register()` is provided by `@lwc/engine`. This is, on itself, complex and confusing. Additionally, there is another `register()` method from `wire-service` that is used by authors to link their wire adapters with their adapter ID (identity). This process also posses a limitation, and unnecessary dependency, making adapters to tied to LWC.
+
+Additionally, there are various situations where the wired field or method is not behaving correctly, the most notable example is when a configuration value that uses a member expression might not trigger the update on the config. (e.g. `wire(foo, { x: '$foo.bar' }) data`, if `foo.bar` changes, config is not updated). This is because the wire decorator is not relying on the reactivity system used by LWC, and instead it relies on getter and setters that are slower, intrusive, complex and do not cover the whole spectrum of mutations that could occur on a component.
+
+Finally, keeping the wire service tied to LWC means that when needed, wire adapters will not be very useful beyond LWC, when in reality their are just not tied to the component system.
+
+# Goals
+
+The primary goal of this RFC is to decouple the wire service from the LWC and from the LWC Wire Decorator implementation.
+
+As a secondary goal is to embrace reactivity for the configuration payload and the wired method in LWC.
+
+A third goal is to support the provision of wire adapters via wire service on any object, whether it is LWC component or not.
+
+# No-goals
+
+* This reform does not change the wire decorator syntax.
+* This reform does not change the wire adapter API for lightning platform (we should be able to keep that intact).
+
+# Proposal
+
+This reform is focused on the refactor of the wire decorator code, and the wire-service code. As part of the separation process, there are certain responsibilities that must be well defined:
+
+## Responsibilities of the wire decorator
+
+* To install a prototype descriptor to handle the wired field value from the `vm` of the component.
+* To create an instance of an adapter and link it to the host.
+* To signal to the adapter instance when the component is connected or disconnected.
+* To signal to the adapter instance when the config have changed by providing the new config object.
+* To extract the config value from the host object by relying on the compiler's wire metadata.
+* To signal to the adapter instance when a context is available by providing the new context value. 
```

Comment:
How is this point different from responsibility #3?

Reply:
3 is about connect and disconnect, this is about contextual configuration.